types/index.d.ts

```diff
@@ -0,0 +1,151 @@
+import * as stream from 'stream';
+import { EventEmitter } from 'events'
+
+
+export type Callback<V, Err = any> = (err: Err, value: V) => void;
+
+export type CodecTransformer = (buffer: Buffer, callback: () => void) => Buffer; // TODO
```

Comment:
It'd probably be easier to extend if we omit partial types entirely. Can you remove this and all other non-complete types in the file?

Reply:
The `CodecTransformer` definition is almost correct, as far as I can see it, only the callback definition might be lacking an argument. I think, the definition has a certain value, because programmers will provide those functions and thus can be sure they are defined correctly.